Add new basic military mine operations manual and buff the existing one

strategineer opened the pull request:
<!-- HOW TO USE: Under each "#### Heading" below, enter information relevant to your pull request.
Leave the headings unless they don't apply to your PR.

Please read carefully and don't delete the comments delimited by "< !--" and "-- >"
Once a pull request is submitted, automatic stylistic and consistency checks will be performed on the PR's changes.
The results of these can be either seen under the "Files changed" section of a PR or in the check's details.

NOTE: Please grant permission for repository maintainers to edit your PR.  It is EXTREMELY common for PRs to be held up due to trivial changes being requested and the author being unavailable to make them. -->

#### Summary
Balance "Add new basic military mine operations manual and buff the existing one"
<!-- This section should consist of exactly one line, edit the one above.
1. Replace the word "Category" with one of these words: Features, Content, Interface, Mods, Balance, Bugfixes, Performance, Infrastructure, Build, I18N.
2. Replace the text inside the quotes with a brief description of your changes.
Or if you don't want a changelog entry, replace the whole line with just the word "None" (with no quotes).
For more on the meaning of each category, see:
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/CHANGELOG_GUIDELINES.md
If approved and merged, your summary will be added to the project changelog:
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/changelog.txt -->

#### Purpose of change

<!-- With a few sentences, describe your reasons for making this change.  If it relates to an existing issue, you can link it with a # followed by the GitHub issue number, like #1234.  If your pull request *fully* resolves an issue, include the word "Fix" or "Fixes" before the issue number, like: Fixes #xxxx
If there is no related issue, explain here what issue, feature, or other concern you are addressing.  If this is a bugfix, include steps to reproduce the original bug, so your fix can be verified. -->


Fixes #63569


#### Describe the solution

<!-- How does the feature work, or how does this fix a bug?  The easier you make your solution to understand, the faster it can get merged. -->

Buff military mine ops manual to make it as useful as it should be by:
- Increasing the traps/devices level it can teach from 3 to 5 (with a minimum devices level of 3 needed to read it)
- Adding two unique practice recipes granted upon reading the manual that allow the player character to gain the required proficiencies to set and disarm booby traps and mines (two recipes also learned by reading the same manual) (mostly?) safely.
- Adding another more basic version of this military trap manual to cover devices level 0 to 3 and act as a primer for the more advanced manual.

NB: The practice recipes require similar components to those needed by the booby trap and mine recipes excluding the explosive components for both realism (practicing with actual explosives seems like a bad idea) and balance (requiring an exorbitant amount of explosive components to do these practice recipes wouldn't be fun IMO because that would leave the player with less explosive components to make actual explosives with their newly gained skills) purposes.

#### Describe alternatives you've considered

<!-- Explain any alternative solutions, different approaches, or possibilities you've considered using to solve the same problem. -->

None but I'm open to suggestions. It's hard for me to test this kind of balance change in a vacuum given my relative inexperience with the game (for now). One book that can bring a character from devices level 0 to 7 might be busted.

I thought about having a single recipe instead of two but putting three proficiencies into one practice recipe seemed to make the practice recipe take too long and made it harder to figure out a reasonable set of training components.

Three recipes might be overkill (eg. basics, setting, disarming) but let me know!

#### Testing

<!-- Describe what steps you took to test that this PR resolved the bug or added the feature, and what tests you performed to make sure it didn't cause any regressions.  Also include testing suggestions for reviewers and maintainers. -->

1. I spawned in both books on a fresh character.
2. Read the basic book until I reached devices level 3, ensuring that the first military trap recipe was learned upon reading the book the first time.
3. Placed some booby traps.
5. Spawned the needed ingredients and started continuously doing the newly added "military traps" training recipe whilst evaluating the resource expenditure, the rate at which the basic trap proficiency was gained, the time taken, etc. until I had gained the proficiency.
6. Read the advanced book until I reached devices level 6, ensuring that the advanced military trap recipe was learned upon reading the book the first time.
7. Did the same for the "military traps (advanced)" recipe until I had gained the next two trap proficiencies.
8. Spawned a bunch of booby traps and mines, placed them and disarmed them safely dozens of times (I'm not sure if theoretically disarming a mine at this point would possibly have it blow up in your face but if so, it seems like it would be quite rare).
9. Repeated this process a bunch until I got something that felt "right".

strategineer committed:
Buff military mine ops manual to make it as useful as it should be given its description

... by increasing the traps/devices level it can teach from 3 to 5 and by adding two unique practice recipes granted upon reading the manual that allow the player character to gain the required proficiencies to set and disarm booby traps and mines (two recipes also learned by reading the same manual) (mostly?) safely.

Fixes #63569

Maleclypse commented on data/json/items/book/traps.json:
```diff
@@ -74,7 +74,7 @@
     "symbol": "?",
     "color": "green",
     "skill": "traps",
-    "max_level": 3,
```
I would suggest that if you are making the max 5 you need a minimum level of around 3.  Which might mean there needs to be a book that handles the lower level of traps.

strategineer replied:
Makes sense, done!

"The Modern Trapper" book might already fill that niche? It sort of makes sense to start with making bear traps and then getting into explosive traps.

strategineer replied:
Although I'll need to tweak the skill values in the practice recipes to account for this... And do some more testing :)

strategineer replied:
I've added a basic military trap book!

strategineer committed:
Set required devices/traps level for military mine ops book to 3

... to balance out it being able to bring the character to level 5

PatrikLundell commented:
While The Modern Trapper might fill the the low level skill needs from a technical point of view, military organizations tend to train people from scratch (or at least be set up to be able to do so), so a military basic training manual ought to exist.

Zireael07 commented:
Mines aren't something you're training people from scratch on, though? That's something reserved for specialized forces

PatrikLundell commented:
I claim that yes, you need to train mine handling from scratch at some stage. Otherwise the special forces would have to only train people who somehow had gotten the basics from outside the military, complete with potential bad practices (probably hunters and farmers, as military recruits tend to be young, and thus rarely would have a prior career in using explosives for construction, mining, etc.). I would expect regular forces to get some basic training in explosives detection and brute force elimination (by setting them off/destroying them from a safe distance), but there would have to be additional training for people intended to actually be able to disarm explosive devices, and that would likely have to be done in several steps.

strategineer committed:
Add basic military trap book to go with the buffed military trap book

strategineer committed:
Set max skill from advanced military traps practice to 6 because that's all you need to safely disarm traps (combined with the 3 trap proficiencies)

strategineer committed:
Remove obsolete 'manual' postfix to name of advanced military trap book to match conventions

strategineer commented:
Thanks for the help y'all.

I've:
- Added a basic version of this military trap book.
- Lined up the values for the intelligence, max_level, time etc. for the military trap books based on the other books in the same file.
- Added the new basic military trap practice recipe to the basic book (the advanced book contains both).
- Set the max devices level that can be trained by the advanced recipe to 6 instead of 7 because that seems like it's enough to safely disarm land mines and lines up with max level you can gain from the book.

strategineer committed:
Fix the military traps practice recipes being autolearned on character creation

strategineer commented:
I tested this some more, should be good to go. Let me know if there's anything else!
